### ISSUE TITLE
frappe.defaults.get_user_default("Company") not working with multi-company

### ISSUE THREAD

**Issue by chdecultot** [frappe/erpnext#15661]
Hi,

It seems that there is an issue with the following method:

`frappe.defaults.get_user_default("Company")`

I have two companies in ERPNext:
- If I don't assign an user to a company, then it correctly gets the default company.  

- But if I create an user permission to assign my user to a particular company, then it returns `undefined`.

It creates a few UX issues as it is widely used in ERPNext.

Have a nice day!

**Comment by kennethsequeira** [comment]
@chdecultot 

I believe this results in some annoying pop-up messages, correct?

Can you provide any steps/scenarios to test this issue out?

**Comment by chdecultot** [comment]
Hi @kennethsequeira,

No actually it doesn't result in any popup in the cases I have tested.

The case I have noticed is the following:

- In opportunities, if not company is selected, then the user default company is assigned automatically.
If there is no user default company, then the system default company is assigned.

In a case with two companies, if my user is assigned to company B and company A is the default company for the system, then the opportunity is assigned to company A, whereas it should have been assigned to company B.

I hope it is clear enough, let me know if your have any additional question :wink:
